python/tank/deploy/tank_commands/desktop_migration.py

```diff
@@ -0,0 +1,93 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import textwrap
+
+from .action_base import Action
+from . import console_utils
+from ...platform import constants
+
+_MESSAGE = ("This will migrate the Shotgun Desktop configuration away from the 'Template "
+            "Project'. Before running this, make sure all your users have installed version 1.2.0 or "
+            "greater of the Shotgun Desktop or that they are running version 1.2.0 or greater of the Startup framework. "
+            "You can see which versions you are running in the Shotgun Desktop's About Box.")
+
+
+class DesktopMigration(Action):
+    """
+    Action that migrates the Shotgun Desktop away from the Template Project.
+    """
+    def __init__(self):
+        """
+        Constructor
+        """
+        Action.__init__(self,
+                        "migrate_desktop",
+                        Action.TK_INSTANCE,
+                        "Migrates Shotgun Desktop away from the Template Project.",
+                        "Admin")
+
+    def run_interactive(self, log, args):
+        """
+        Tank command accessor (tank migrate_desktop)
+        Retrieves the pipeline configuration from Shotgun and unassigns it from the Template Project.
+        """
+
+        log.info("Retrieving pipeline configuration from Shotgun...")
+
+        pc = self.tk.pipeline_configuration
+
+        # Fetch the project id from the actual pipeline configuration.
+        sg_pc = self.tk.shotgun.find_one(
+            constants.PIPELINE_CONFIGURATION_ENTITY,
+            [["id", "is", pc.get_shotgun_id()]],
+            ["project.Project.name"]
+        )
+
+        # If the pipeline configuration doesn't exist anymore, abort.
+        if not sg_pc:
+            log.error("Pipeline configuration '%d' doesn't seem to exist anymore." % pc.get_shotgun_id())
+            return
+
+        # If a pipeline configuration is not assigned to the Template Project, abort.
+        project_name = sg_pc["project.Project.name"]
+        if project_name and project_name != "Template Project":
+            log.error(
+                "Migration is possible only from the project named \"Template Project\". This "
+                "configuration is for project '%s'" % project_name)
+            return
+
```

Comment:
Should print out details of the PC we found (linked project id, PC id, PC name) just for debuggability.
Should also reaffirm the warning above since we're about to prompt yes no.